From 1c75ea0332054c420efd676c147e45eee43e0118 Mon Sep 17 00:00:00 2001
From: Jonathan Liu <net147@gmail.com>
Date: Thu, 6 Jun 2013 12:42:13 +1000
Subject: [PATCH] Don't overwrite QMAKE_QT_CONFIG with empty value

If the mkspec sets QMAKE_QT_CONFIG, QMAKE_QT_CONFIG may be overwritten
with an empty value from .qmake.cache. Avoid this by first checking
if the value from .qmake.cache is not empty before assigning it to
QMAKE_QT_CONFIG.

Upstream-Status: Submitted
https://codereview.qt-project.org/#change,58109

Change-Id: I95fa630139b8798156a2fb15d0dde630a0a53a0a
Signed-off-by: Jonathan Liu <net147@gmail.com>
---
 mkspecs/features/qt_config.prf | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/mkspecs/features/qt_config.prf b/mkspecs/features/qt_config.prf
index fc14cdd..e059b7e 100644
--- a/mkspecs/features/qt_config.prf
+++ b/mkspecs/features/qt_config.prf
@@ -1,7 +1,11 @@
 # This file is loaded by the mkspecs, before .qmake.cache has been loaded.
 # Consequently, we have to do some stunts to get values out of the cache.
 
-exists($$_QMAKE_CACHE_):QMAKE_QT_CONFIG = $$fromfile($$_QMAKE_CACHE_, QMAKE_QT_CONFIG)
+exists($$_QMAKE_CACHE_) {
+   qdd = $$fromfile($$_QMAKE_CACHE_, QMAKE_QT_CONFIG)
+   !isEmpty(qdd): QMAKE_QT_CONFIG = $$qdd
+   unset(qdd)
+}
 isEmpty(QMAKE_QT_CONFIG)|!exists($$QMAKE_QT_CONFIG) {
    qdd = $$QT_BUILD_TREE
    isEmpty(qdd):exists($$_QMAKE_CACHE_): qdd = $$fromfile($$_QMAKE_CACHE_, QT_BUILD_TREE)
-- 
1.8.2.3

